halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/DeploymentConfigurationValidator.java

```diff
@@ -76,11 +76,24 @@ public void validate(ConfigProblemSetBuilder p, DeploymentConfiguration n) {
       return;
     }
 
-    boolean isReleased = versions.getVersions()
+    Optional<Versions.Version> releasedVersion = versions.getVersions()
         .stream()
-        .anyMatch(v -> Objects.equals(v.getVersion(), version));
+        .filter(v -> Objects.equals(v.getVersion(), version))
+        .findFirst();
 
-    if (!isReleased) {
+    boolean isReleased = releasedVersion.isPresent();
+
+    String runningVersion = versionsService.getRunningHalyardVersion();
+    boolean halyardSnapshotRelease = runningVersion.endsWith("SNAPSHOT");
+
+    if (isReleased) {
+      String requiredHalyardVersion = releasedVersion.get().getHalyardVersion();
+      if (!halyardSnapshotRelease
+          && !StringUtils.isEmpty(requiredHalyardVersion)
+          && Versions.lessThan(runningVersion, requiredHalyardVersion)) {
+        p.addProblem(Problem.Severity.ERROR, "Halyard version \"" + runningVersion + "\" is less than Halyard version \"" + requiredHalyardVersion + "\" required for Spinnaker \"" + version + "\"");
+      }
+    } else {
```

Comment:
Consider removing this `else` and `return`ing at the end of the conditional block. It reads easier to not have a whole bunch of stuff in an else block.